### PR TITLE
Fix randomize null join key optimizer

### PR DESCRIPTION
**Pull request by feilong-liu**
Join output has a restriction that output from left input should be before output from right input. Fix the randomize null join key optimizer here to keep this order.

This is not a problem most time, since the randomized join key is often only used in the join equi condition only, and even if we add it into output, it will be pruned later. Another reason is that, some other optimization rules which touch the join node will recreate the join node, and rearrange the join output (like what I added in the unit test in this PR).

### Test plan - (Please fill in how you tested your changes)

Add unit tests.

```
== RELEASE NOTES ==

General Changes
* Fix join output in RandomizeNullKeyInOuterJoin rule.
```

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java [hunk 1]
```diff
@@ -119,6 +120,36 @@ public void testLeftJoinOnSameKey()
                 false);
     }
 
+    @Test
+    public void testLeftJoinOnSameKeyJoinAsRightSideInput()
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java:124]: This test will fail without the changes in this PR, and succeed after making the changes.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java [hunk 2]
```diff
@@ -119,6 +120,36 @@ public void testLeftJoinOnSameKey()
...
+    {
+        assertPlan("select * from partsupp ps left join (select p.name, l.orderkey, l.partkey as partkey from part p left join lineitem l on p.partkey = l.partkey) pl on ps.partkey = pl.partkey",
+                Session.builder(getSessionAlwaysEnabled()).setSystemProperty(OPTIMIZE_HASH_GENERATION, "false").build(),
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java:127]: To disable optimization hash generation, as this optimizer will recreate the join node and rearrange the join output, hence cover the problem here.

**Review by pranjalssh — APPROVED** [review]
LGTM

**Comment by rohanpednekar** [comment]
@feilong-liu  can you confirm if default is false for above property?

**Pull request merged**